### PR TITLE
Connection auto recover

### PR DESCRIPTION
**Pull request by seregazhuk**

### CODE REVIEW

File: src/Factory.php [hunk 1]
```diff
@@ -13,40 +11,31 @@
...
+     * @param LoopInterface $loop
+     * @param $address
+     * @return Connection
      */
-    private function createProtocolParser()
+    protected static function createConnection(LoopInterface $loop, $address)
     {
-        return new Parser(new RequestFactory(), new ResponseFactory());
+        return new Connection($address, new Connector($loop));
     }
```
Comment (kelunik) [inline src/Factory.php:40]: I'd just hardcode this and accept a `Connector` as constructor argument (if you want).

**Comment by kelunik** [comment]
@seregazhuk I don't think you need factories for value objects, such as request and response.

**Comment by seregazhuk** [comment]
@kelunik I wanted to isolate logic for creating different types of requests and responses from the protocol parser. Is it better to move this logic to protocol parser? or don't inject these factories and make them static?

**Comment by kelunik** [comment]
If you make them static, you can also make them simple methods. I see no problem with having that logic in the parser if it's that simple and only a mapping of the parse result into an object for better accessibility.

**Pull request merged**